src/classes/chart/Song.as

```diff
@@ -70,21 +71,27 @@ package classes.chart
         private var localFileData:ByteArray = null;
         private var localFileHash:String = "";
 
-        public function Song(song:Object, isPreview:Boolean = false):void
+        public function Song(songInfo:SongInfo, isPreview:Boolean = false):void
         {
-            this.entry = song;
-            this.id = song.level;
+            this.songInfo = songInfo;
+            this.id = songInfo.level;
             this.preview = isPreview;
-            this.type = (song.type != null ? song.type : NoteChart.FFR);
-            this.chartType = song.type || NoteChart.FFR_LEGACY;
+            this.type = (songInfo.chartType != null ? songInfo.chartType : NoteChart.FFR);
```

Comment:
Could be changed to `this.type = songInfo.chartType || NoteChart.FFR;` to match the assignment chartType.